src/constructors/regular.c

```diff
@@ -420,6 +420,81 @@ igraph_error_t igraph_tree(igraph_t *graph, igraph_integer_t n, igraph_integer_t
     return IGRAPH_SUCCESS;
 }
 
+igraph_error_t igraph_tree_from_parent_vector(igraph_t *graph, igraph_vector_int_t* parents,
+                igraph_tree_mode_t type) {
+
+    igraph_vector_int_t edges = IGRAPH_VECTOR_NULL;
+    igraph_vector_int_t visited = IGRAPH_VECTOR_NULL;
+    igraph_integer_t idx = 0;
+    igraph_integer_t to = 1;
+
+    if (igraph_vector_int_empty(parents)) {
```

Comment:
The number of vertices in the graph will be the same as the length of the `parents` vector. Thus, an empty parents vector can represent the null graph, which is a forest, thus a valid result.